### ISSUE TITLE
A proper way to choose alternative KAF shapes is needed

### ISSUE THREAD

**Issue by munzirtaha** [DigitalKhatt/madinafont#18]
Applying `ss02` for **elongated KAF** converts certain cases but not others, e.g.
كل كله
![ss02](https://github.com/DigitalKhatt/madinafont/assets/2936425/bc0036bd-ebf9-4f0a-b640-c517ace6847e)

**Comment by aminanan** [comment]
This was experimental and is no longer used. How to be able to choose alternatives is still under development and not yet finalized.

**Comment by munzirtaha** [comment]
I understand but let's leave this bug open till there is a way to at least change the KAF shape properly.

**Comment by aminanan** [comment]
Can you try feature cv03 ? You have to apply it also to the surrounding character if it is a ligature.

**Comment by munzirtaha** [comment]
Yes, it worked with the example I mentioned but didn't work after THAL e.g.
ءَآلذَّكَرَيْنِ

**Comment by aminanan** [comment]
Here is the result for ۝ ءَآلذَّكَرَيْنِ ۝ كله ۝ كل  with this style `font-feature-settings: 'cv03' 1`

![Capture](https://github.com/user-attachments/assets/6e49999e-092e-464c-8da8-7cb2a4903e5d)

**Comment by munzirtaha** [comment]
Thanks, it's fixed now. Another position which is not fixed yet is
مُهْلِكِي
وَأَبْكَىٰ

**Comment by aminanan** [comment]
The alternate Kaf before final Yeh exists only once in the word بِتَارِكِيٓ, Surah Hud Ayat 53. So I didn't add this substitution so that it doesn't apply during justification.

To not affect the justification I added a new feature `ss01` where you can choose alternate Kaf before final Yeh. You also need to apply this feature to the letter Yeh, otherwise you will get a bad shape. Here is the result for the text ۝ وَأَبْكَىٰ ۝ مُهْلِكِي ۝ ءَآلذَّكَرَيْنِ ۝ كله ۝ كل with this style `font-feature-settings: 'ss01' 1;`

![Capture](https://github.com/user-attachments/assets/bfebc821-29f3-4669-a881-4c7b263f3fb1)

**Comment by munzirtaha** [comment]
Actually, I'd love to see it used during justification. Sorry, but I couldn't see a purpose for this distinction. A single use is sufficient to demonstrate its correctness, and that it can be utilized for justification and variety. When apps are sophisticated enough, one can choose which alternates he prefers to be applied automatically or have a priority list, but these will be personal preferences.

قمت الآن بمراجعة بعض كراسات الخط العربي وأمشاق كبار خطاطي العالم ومصاحفهم ووجدتهم يستخدمون الكاف الطويلة مع الياء المتطرفة لذلك لا أرى سببا لهذا التفريق، فإن كنت تعلم من استقبح هذا فأخبرني فلست من أهل الفن لكن هذا ما بدا لي والله أعلم


هناك ملاحظة في كلمة "مُهْلِكِي" بالكاف العادية، الامتداد بين اللام والكاف أطول من اللازم مقارنة بنسخ المصحف

![مهلكي](https://github.com/user-attachments/assets/403df942-0c40-4d34-ab1d-78b92353f4f3)

**Comment by aminanan** [comment]
I just wanted to mention that I did not apply the Kaf alternative before the final Yeh for justification since one case is not enough to deduce the rule which must take into account the other possibilities of justification.  So I added ss01 so that it doesn't interfere with the current justification algorithm until further analysis.

For the Kaf issue mentioned above, please create a new issue as it is a separate problem not related to this one.

**Comment by munzirtaha** [comment]
Ok, I will close this issue for now. Thanks.
